tasks/broad/UltimaGenomicsGermlineFilteringThreshold.wdl

```diff
@@ -0,0 +1,401 @@
+            flow_order = flow_order
+
+    }
+
+    call EvaluateResults {
+        input:
+          h5_input = CompareToGroundTruth.compare_h5,
+          base_file_name = base_file_name,
+          disk_size = 10,
+          score_key = score_key
+    }
+
+    scatter (idx in range(length(input_vcf))) {
+        call HardThresholdVCF {
+            input:
+              thresholds = EvaluateResults.thresholds_report,
+              input_vcf = AnnotateSampleVCF.output_vcf_file[idx],
+              input_vcf_index = AnnotateSampleVCF.output_vcf_index[idx],
+              score_key = score_key,
+              output_basename = base_file_name,
+              disk_size = 3*ceil(size(AnnotateSampleVCF.output_vcf_file[idx], "GB")) + 14
+        }
+    }
+
+    output {
+        Array[File] output_vcf = HardThresholdVCF.output_vcf
+        Array[File] output_vcf_index = HardThresholdVCF.output_vcf_index
+        File eval_report_h5 = EvaluateResults.eval_report_h5
+        File thresholds_report = EvaluateResults.thresholds_report
+    }
+}
+
+task ExtractSample {
+    input {
+        String sample_name
+        File input_vcf
+        File input_vcf_index
+        String docker = "us.gcr.io/broad-gotc-prod/imputation-bcf-vcf:1.0.6-1.10.2-0.1.16-1659548257"
+    }
+    String output_vcf = basename(input_vcf, ".vcf.gz") + ".~{sample_name}.vcf.gz"
+    command <<<
+        set -eo pipefail
+
+        bcftools view -s ~{sample_name} ~{input_vcf} -o ~{output_vcf} -O z
+        bcftools index -t ~{output_vcf}
+    >>>
+    output {
+        File output_vcf_file = "~{output_vcf}"
+        File output_vcf_index = "~{output_vcf}.tbi"
+    }
+    runtime {
+        memory: "8GB"
+        disks: "local-disk " + (ceil(size(input_vcf, "GB")) * 2 + 10) + " HDD"
+        docker: docker
+        cpu: 1
+    }
+}
+
+task FilterSampleVCF{
+    input{
+        File input_vcf
+        String docker = "gcr.io/terra-project-249020/jukebox_vc:test_jc_optimize_3d7509"
+    }
+
+    String output_vcf = basename(input_vcf, ".vcf.gz") + ".filter_unused.vcf.gz"
+
+    command <<<
+        set -e
+        source ~/.bashrc
+        conda activate genomics.py3
+        bcftools view -a -i 'GT[*]="alt"' ~{input_vcf} -o ~{output_vcf} -O z 
+        bcftools index -t ~{output_vcf}
+    >>>
+
+    output {
+        File output_vcf_file = "~{output_vcf}"
+        File output_vcf_index = "~{output_vcf}.tbi"
+    }
+
+    runtime {
+        memory: "8GB"
+        disks: "local-disk " + (ceil(size(input_vcf, "GB")) *2 + 10) + " HDD"
+        docker: docker
+        cpu: 1
+    }
+}
+
+task FilterSymbolicAlleles {
+    input {
+        String base_file_name
+        File input_vcf
+        File input_vcf_index
+        String docker = "gcr.io/terra-project-249020/jukebox_vc:test_jc_optimize_3d7509"
+    }
+
+    String output_vcf_name = "~{base_file_name}" + ".vcf.gz"
+    command <<<
+        set -e
+        source ~/.bashrc
```

Comment:
just wondering why this line is necessary?

Reply:
huh, the `.bashrc` sets up conda, so I think it's needed for `conda activate genomics.py3`, but this task (and some of the others) don't use python so I think it can be removed. I'll clean this up.

Reply:
actually I was wrong, in this docker gatk and bcftools are managed by conda. I'll add a comment in the code.